### PR TITLE
chore(transformation): setting default timeout for user transformation

### PR DESCRIPTION
**Pull request by Jayachand**
## Description of the change

> Setting 10 min as global default timeout to accommodate fetch api calls
> In hosted, this env will be set to a lower threshold say 10 sec

https://www.notion.so/rudderstacks/Add-global-timeout-env-for-fetch-calls-ecbdb190732d48498a88f20e0b8ef263

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by sivashanmukh — DISMISSED** [review]

File: util/customTransformer-v1.js [hunk 1]
```diff
@@ -30,9 +34,23 @@ async function transform(isolatevm, events) {
...
-  return executionPromise.catch(e => {
-    throw new Error(e);
-  });
+
+  let result;
+  try {
+    const timeoutPromise = new Promise(resolve => {
+      const wait = setTimeout(() => {
+        clearTimeout(wait);
+        resolve("Timedout");
```
Comment (koladilip) [inline util/customTransformer-v1.js:43]: it is possible to reject directly from here right? Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race
if we do this, we may not need to wrap into try-catch block

Comment (Jayachand) [reply util/customTransformer-v1.js]: made the changes

File: util/customTransformer-v1.js [hunk 2]
```diff
@@ -35,22 +36,17 @@ async function transform(isolatevm, events) {
...
-  }
-  return result;
+  let setTimeoutHandle;
+  const timeoutPromise = new Promise((_, reject) => {
+    setTimeoutHandle = setTimeout(() => {
+      reject(new Error("Timed out"));
+    }, userTransformTimeout);
+  });
+  return Promise.race([executionPromise, timeoutPromise])
+    .catch(e => {
```
Comment (koladilip) [inline util/customTransformer-v1.js:46]: this is catch required?

Comment (Jayachand) [reply util/customTransformer-v1.js:46]: It shows `unhandledpromiserejectionwarning` warning during unit tests. added for readibility

**Review by koladilip — APPROVED** [review]

File: util/customTransformer-v1.js [hunk 3]
```diff
@@ -5,6 +5,11 @@ const { getFactory } = require("./ivmFactory");
 const { getMetadata } = require("./../v0/util");
 const logger = require("../logger");
 
+const userTransformTimeout = parseInt(
+  process.env.USER_TRANSFORM_TIMEOUT || "600000",
```
Comment (saikumarrs) [inline util/customTransformer-v1.js:9]: I'd suffix it with `_MS` for improving readability.

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**